recipes/dlib/bld.bat

```diff
@@ -0,0 +1,47 @@
+@echo ON
+
+mkdir build
+cd build
+
+rem The Python lib has no period in the version string, so we remove it here.
+set PY_VER_NO_DOT=%PY_VER:.=%
+rem Required as the png locator does some processing that chokes on \
+set "PNG_LIBRARY=%LIBRARY_LIB%\libpng16.lib"
+set "PNG_LIBRARY=%PNG_LIBRARY:\=/%"
+
+rem There is a bug whereby linking to jpeg causes a link crash, so we allow
+rem dlib to build it
+cmake ..\tools\python -LAH -G"NMake Makefiles"              ^
+-DCMAKE_BUILD_TYPE="Release"                                ^
+-DCMAKE_PREFIX_PATH="%LIBRARY_PREFIX%"                      ^
+-DBoost_USE_STATIC_LIBS=0                                   ^
+-DBoost_USE_STATIC_RUNTIME=0                                ^
+-DBOOST_ROOT="%LIBRARY_PREFIX%"                             ^
+-DBOOST_INCLUDEDIR="%LIBRARY_INC%"                          ^
+-DBOOST_LIBRARYDIR="%LIBRARY_LIB%"                          ^
+-DPYTHON3=%PY3K%                                            ^
+-DPYTHON_LIBRARY="%PREFIX%\libs\python%PY_VER_NO_DOT%.lib"  ^
+-DPYTHON_INCLUDE_DIR="%PREFIX%\include"                     ^
+-DDLIB_LINK_WITH_SQLITE3=0                                  ^
+-DDLIB_PNG_SUPPORT=1                                        ^
+-DPNG_INCLUDE_DIR="%LIBRARY_INC%"                           ^
+-DPNG_PNG_INCLUDE_DIR="%LIBRARY_INC%"                       ^
+-DPNG_LIBRARY=%PNG_LIBRARY%                                 ^
+-DZLIB_INCLUDE_DIRS="%LIBRARY_INC%"                         ^
+-DZLIB_LIBRARIES="%LIBRARY_BIN%\zlib.dll"                   ^
+-DDLIB_JPEG_SUPPORT=1                                       ^
```

Comment:
[L503](https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.5184/job/iiixxpbb1ta6fe7m#L503) suggests it is indeed set (plus it actually [builds jpeg](https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.5184/job/iiixxpbb1ta6fe7m#L803). Were you always setting `CXX_FLAGS` on Windows to add the defines (rather than `add_definitions`)?